### PR TITLE
ironic: Expose pxe_append_params as attribute + migration fix

### PR DESCRIPTION
**Pull request by skazi0**
pxe_append_params option is used to add extra kernel parameters to IPA runs.
Example use:
`sshkey="<public key>"` param gives ssh access to running IPA image.
`coreos.autologin` enables console auto login (CoreOS based image)
`ipa-debug=1` enables debug logging in IPA
Great for troubleshooting.

EDIT: additional commit fixes small bug causing migrations to fail.

### CODE REVIEW

**Comment by skazi0** [comment]
@nicolasbock the line you linked does provide the logger parameter.
The problem I had during migrations had this trace:

```
wrong number of arguments (0 for 1)
/opt/dell/crowbar_framework/app/models/ironic_service.rb:19:in `initialize'
/opt/dell/crowbar_framework/lib/schema_migration.rb:63:in `new'
/opt/dell/crowbar_framework/lib/schema_migration.rb:63:in `block in run_for_bc'
/opt/dell/crowbar_framework/lib/schema_migration.rb:49:in `run_for_bc'
/opt/dell/crowbar_framework/lib/tasks/crowbar.rake:30:in `block (3 levels) in <top (required)>'
/opt/dell/crowbar_framework/lib/tasks/crowbar.rake:29:in `each'
/opt/dell/crowbar_framework/lib/tasks/crowbar.rake:29:in `block (2 levels) in <top (required)>'
```

**Comment by nicolasbock** [comment]
Yes, you are right. Thanks for the backtrace.

File: chef/data_bags/crowbar/migrate/ironic/202_add_pxe_append_params.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+def upgrade(ta, td, a, d)
+  a["pxe_append_params"] = ta["pxe_append_params"]
```
Comment (toabctl) [inline chef/data_bags/crowbar/migrate/ironic/202_add_pxe_append_params.rb]: If we want to backport this, a check if the key is already there is needed.

Comment (skazi0) [reply chef/data_bags/crowbar/migrate/ironic/202_add_pxe_append_params.rb]: I don't think we need to backport this but I added the check just in case.

**Pull request merged**